test/test_methods.py

```diff
@@ -368,3 +371,31 @@ def test_filter_close(self):
         # Must close the underlying file object
         devnull_filter.close()
         self.assertTrue(fd.closed)
+
+
+class MethodsQueueTest(unittest.TestCase):
+
+    def test_queue_min_size(self):
```

Comment:
Not quite sure how to do this. If the exception isn't raised the thread blocks while waiting for data, therefore the test won't complete.

Reply:
I meant that for example `test_queue_min_size()` could be split (and renamed) into two different unit tests: one where things work fine (e.g. we put two elements, so the call to `qsize()` returns the expected value, two)...

```
def test_qsize(self):
    q = methods.Queue()
    q.put_nowait(2)
    q.put_nowait(3)
    self.assertEqual(2, q.qsize())
```
... and another one, where we try to `get()` from an empty queue: the exception is raised but after that `qsize()` still returns the correct value (zero, instead of -1 as happened earlier):

```
def test_qsize_empty_queue(self):
    q = methods.Queue()
    with self.assertRaises(Queue.Empty):
        q.qsize()
    self.assertEqual(0, q.qsize())

```